source/common/http/filter_manager.cc

```diff
@@ -787,12 +822,20 @@ void FilterManager::decodeTrailers(ActiveStreamDecoderFilter* filter, RequestTra
     }
 
     processNewlyAddedMetadata();
+    // Check if the last filter has processed trailers
+    terminal_filter_reached = std::next(entry) == decoder_filters_.end();
 
-    if (!(*entry)->commonHandleAfterTrailersCallback(status)) {
+    if (terminal_filter_reached) {
+      break;
```

Comment:
I don't think we should break here, seems like we could miss a case where data as stopped earlier but trailers went through. I'm not 100% sure.

Reply:
I have removed this part, it was not needed. However the change `if (!(*entry)->commonHandleAfterTrailersCallback(status) && !terminal_filter_reached) {` is needed. I think this is pre-existing bug as the loop never reached the end, since the terminal filter always returns FilterTrailersStatus::StopIteration

So it does need the same check as in the decodeData and decodeHeaders to ignore the return for the terminal filter status.